### PR TITLE
Update modular-code.md

### PR DESCRIPTION
**Pull request by ErinMB**
Erin, Tim & Keerthi revised various sections. Team - please review!

### CODE REVIEW

File: modular-code.md
Comment (rrrutledge) [inline modular-code.md]: Not a huge deal in this case, but recommend [putting each sentence on its own line](https://github.com/paypal/InnerSourcePatterns/pull/96).

Comment (rrrutledge) [inline modular-code.md]: > Education might be needed.

Feels like a potential solution rather than a force?

File: modular-code.md
Comment (rrrutledge) [inline modular-code.md]: I don't understand the connection between these ideas.  Modular code seems even _more_ useful with rotating team members so that no one has to keep the whole monolith in their head?

File: modular-code.md
Comment (rrrutledge) [inline modular-code.md]: Needs bullets or something else to make each name appear on its own line.

**Review by rrrutledge — APPROVED** [review]
Some comments that I think make sense to address, but 
also _approving_ as the submitted document seems to be in as-good or better state than the `master` document.

File: modular-code.md
Comment (NewMexicoKid) [reply modular-code.md]: @rrrutledge, the experience shared was that in a team with frequent turnover, the emphasis (sadly) wasn't on producing more maintainable code but in struggling to meet deadlines (no bandwidth for making their code more modular).

File: modular-code.md
Comment (NewMexicoKid) [reply modular-code.md]: Good catch (and education is indeed part of the solution). We can probably delete the sentence (Education might be needed).

File: modular-code.md
Comment (rrrutledge) [reply modular-code.md]: Got it.

Comment (rrrutledge) [reply modular-code.md]: I'd say that's less about turnover of team members than just under short-term time pressure in-general.  Could be for a variety of reasons but we'd expect to see that result no matter what outside force is driving that pressure.

**Comment by spier** [comment]
I changed the headline formatting of this file as part of https://github.com/InnerSourceCommons/InnerSourcePatterns/pull/131. So if that branch get merged, we will likely see a bunch merge conflicts with this branch.

**Comment by rrrutledge** [comment]
Oh well - this PR is nearly two years old 🤷‍♂

**Comment by lenucksi** [comment]
I've merged the `master` branch into to pull request to enable it to merge cleanly. Luckily @spier's great re-sorting pull request had this sorted into `1-initial` already. 
I'd love to merge this but would appreciate if @NewMexicoKid and @rrrutledge could have a look into the open discussions and resolve them if ready or see into getting them resolved.

**Comment by rrrutledge** [comment]
Great!  Will take a look.

**Comment by rrrutledge** [comment]
None of my discussions are resolved.  I already signed off, though.  I don't think that my comments should block merge.

File: patterns/1-initial/modular-code.md
Comment (gruetter) [inline patterns/1-initial/modular-code.md]: Just to clarify: modularization might not be a **priority for developers,** if they think they are going to leave the team, soon? I would argue it's definitively a priority from an organisations perspective, because modular code should be more understandable and maintainable and that in turn helps when new team members are onboarded.

File: patterns/1-initial/modular-code.md
Comment (gruetter) [inline patterns/1-initial/modular-code.md:35]: I believe this also needs some clarification.

File: patterns/1-initial/modular-code.md
Comment (gruetter) [inline patterns/1-initial/modular-code.md]: How about we combine this bullet point and the next by saying: "modular code facilitates division of labour"?

File: patterns/1-initial/modular-code.md
Comment (gruetter) [inline patterns/1-initial/modular-code.md]: I'm guessing there's a **for** missing: and drive acceptance **for** the cost of modularization

**Review by gruetter — APPROVED** [review]
I think a couple of explanations are required. I'll approve, trusting that the shepherd of this PR will consider my suggestions.

File: patterns/1-initial/modular-code.md
Comment (spier) [reply patterns/1-initial/modular-code.md]:
```suggestion
* Acknowledge and drive acceptance for the cost of modularization. Build time into the release schedule for this.
```

File: patterns/1-initial/modular-code.md
Comment (spier) [inline patterns/1-initial/modular-code.md]: I don't understand what "Pattern Idea" is meant to say. as part of "Resulting Context".
Further I don't know why "Pattern Idea" is formatted this way.

At least for the formatting there could be a fix:
Instead of an extra headline, we could make "Pattern Idea" bold instead? 
That. way it doesn't break the formatting of the pattern template any more.

See below:

```suggestion
**Pattern Idea**: Time is spent making the shared code modular so it can be reused.  
```

File: patterns/1-initial/modular-code.md
Comment (spier) [inline patterns/1-initial/modular-code.md]: Fixes spacing.

```suggestion
## Known Instances

Elements of the resolution have been proven by various companies.

```

File: patterns/1-initial/modular-code.md
Comment (spier) [reply patterns/1-initial/modular-code.md]: I like it.

I added in your suggestion, while steal using part of the original text above as part of the motivation.

```suggestion
    - Modular code facilitates division of labour (as required for work in larger teams, or as an Open Source project)
```

**Review by spier — APPROVED** [review]
Given that this PR is open for 2.5 years already, I suspect that it might not have a shepherd any more. 😢 

Therefore my suggestion would be for anybody of any Trusted Committer here that is still part of the conversation to resolve the comments to the best of their knowledge, and then merge the PR. 

That way we capture at least some of the value that this PR, without keeping it open for an unnecessarily long amount of time.

Once the PR is merged, others can take this pattern forward, as it is currently still in "Initial" state i.e. a pretty immature pattern anyways.

File: patterns/1-initial/modular-code.md
Comment (gruetter) [reply patterns/1-initial/modular-code.md]: WDYT, @MaineC and @NewMexicoKid ?

**Review by lenucksi — APPROVED** [review]
This looks fine to me for its current placement in `1-initial`. 

Looking for a reaction to @gruetter's review comments in https://github.com/InnerSourceCommons/InnerSourcePatterns/pull/95#pullrequestreview-465406658 by @NewMexicoKid and @MaineC and will then merge once those are resolved.

File: patterns/1-initial/modular-code.md
Comment (spier) [inline patterns/1-initial/modular-code.md]:
```suggestion
Initial
```

File: modular-code.md
Comment (spier) [reply modular-code.md]: Done.

File: modular-code.md
Comment (spier) [reply modular-code.md]: Resolved.

Comment (spier) [reply modular-code.md]: Removed this part:
> Education might be needed.

... as it was already part of the solution:
> * Offer education. Modular code is well-understood; there is a lot of literature in favor of this.

File: modular-code.md
Comment (spier) [reply modular-code.md]: There are two points that address the feedback by Russ that are already in the pattern:
> * Developers might have no incentive to write modular code (due to their tight schedules and lack of a mandate).
> * If there is frequent turnover of team members, modularization may not be a priority.

I moved those points side-by-side now and will let this be worked out when the patten gets leveled up later (if it does).

**Comment by spier** [comment]
I worked in most of the feedback from this PR, and left the threads open that I could not resolve myself.

Even though there are open questions from @gruetter I am merging this PR now, as it already contains a number of improvements over the pattern currently in the mainline.

**Pull request merged**
